File: cis_interface/examples/maxMsg/src/maxMsgCli.py
import sys
import random
from cis_interface.interface.CisInterface import CisRpcClient, CIS_MSG_BUF


def rand_str(length):
    charset = ("0123456789" +
               "abcdefghijklmnopqrstuvwxyz" +
               "ABCDEFGHIJKLMNOPQRSTUVWXYZ")
    out = ''
    while (len(out) < (length - 1)):
        index = int(random.random() * len(charset))
        out += charset[index]
    if sys.version_info[0] == 3:
        out = out.encode("utf-8")
    return out


msg_size = CIS_MSG_BUF

print("maxMsgCli(P): Hello message size %d." % msg_size)

# Create a max message, send/recv and verify
rpc = CisRpcClient("maxMsgSrv_maxMsgCli", "%s", "%s")

# Create a max message
output = rand_str(msg_size)

# Call RPC server
ret, input = rpc.rpcCall(output)
if not ret:
    print("maxMsgCli(P): RPC ERROR")
    sys.exit(-1)

# Check to see if response matches
if (input[0] != output):
    print("maxMsgCli(P): ERROR: input/output do not match")
    sys.exit(-1)
else:
    print("maxMsgCli(P): CONFIRM")

# All done, say goodbye
print("maxMsgCli(P): Goodbye!")

File: cis_interface/examples/maxMsg/src/maxMsgCli.m

charset = ['0123456789', ...
	   'abcdefghijklmnopqrstuvwxyz', ...
	   'ABCDEFGHIJKLMNOPQRSTUVWXYZ'];


msg_size = YggInterface('YGG_MSG_BUF');
fprintf('maxMsgCli(M): Hello message size is %d.\n', msg_size);

% Create a max message, send/recv and verify
rpc = YggInterface('YggRpcClient', 'maxMsgSrv_maxMsgCli', '%s', '%s');

% Create a max message
output = randsample(charset, msg_size-1, true);

% Call RPC server
[flag, vars] = rpc.call(output);
if (~flag)
  error('maxMsgCli(M): RPC ERROR');
end;

% Check to see if response matches
if (vars{1} ~= output)
  error('maxMsgCli(M): ERROR: input/output do not match');
else
  disp('maxMsgCli(M): CONFIRM');
end;

% All done, say goodbye
disp('maxMsgCli(M): Goodbye!');
